mail-filter/mapson/files/mapson-3.3-gcc6.patch

```diff
@@ -0,0 +1,285 @@
+--- a/parse-config-file.cpp
++++ b/parse-config-file.cpp
+@@ -55,7 +55,7 @@
+ 
+   std::ifstream file(filename);
+   if (!file)
+-    throw system_error(std::string("parse_config_file() failed to open '") + filename + "'");
++    throw Mapson::system_error(std::string("parse_config_file() failed to open '") + filename + "'");
+ 
+   // Now we read line by line and process each one seperately.
+ 
+--- a/request-confirmation.cpp
++++ b/request-confirmation.cpp
+@@ -157,7 +157,7 @@
+   string filename = config->request_for_confirmation_file;
+   int fd = multi_open(filename, O_RDONLY, S_IRUSR | S_IWUSR);
+   if (fd < 0)
+-    throw system_error(string("Can't open request-mail template file '") + filename + "' for reading");
++    throw Mapson::system_error(string("Can't open request-mail template file '") + filename + "' for reading");
+   fd_sentry sentry(fd);
+ 
+   // Read the file into memory.
+@@ -167,7 +167,7 @@
+   for (rc = read(fd, buffer, sizeof(buffer)); rc > 0; rc = read(fd, buffer, sizeof(buffer)))
+     mail_template.append(buffer, rc);
+   if (rc < 0)
+-    throw system_error(string("Failed to read request-mail template file '") + filename + "' into memory");
++    throw Mapson::system_error(string("Failed to read request-mail template file '") + filename + "' into memory");
+ 
+   // Expand variables in the template.
+ 
+@@ -180,11 +180,11 @@
+   debug(("Executing mail transport agent '%s'.", config->mta.c_str()));
+   FILE* fh = popen(config->mta.c_str(), "w");
+   if (fh == NULL)
+-    throw system_error(string("Can't start MTA '") + config->mta + "'");
++    throw Mapson::system_error(string("Can't start MTA '") + config->mta + "'");
+   if (fwrite(mail_template.data(), mail_template.size(), 1, fh) != 1)
+   {
+     pclose(fh);
+-    throw system_error(string("Failed to pipe to MTA process '") + config->mta + "'");
++    throw Mapson::system_error(string("Failed to pipe to MTA process '") + config->mta + "'");
+   }
+   pclose(fh);
+ }
+--- a/spool.cpp
++++ b/spool.cpp
+@@ -44,13 +44,13 @@
+   info("Spooling e-mail '%s' as '%s'.", config->message_id.c_str(), filename.c_str());
+   int fd = open(filename.c_str(), O_WRONLY | O_CREAT | O_TRUNC, S_IRUSR | S_IWUSR);
+   if (fd < 0)
+-    throw system_error(string("Can't open spool file '") + filename + "' for writing");
++    throw Mapson::system_error(string("Can't open spool file '") + filename + "' for writing");
+   fd_sentry sentry(fd);
+   for (size_t len = 0; len < mail.size(); )
+   {
+     ssize_t rc = write(fd, mail.data()+len, mail.size()-len);
+     if (rc < 0)
+-      throw system_error(string("Failed writing to the spool file '") + filename + "'");
++      throw Mapson::system_error(string("Failed writing to the spool file '") + filename + "'");
+     else
+       len += rc;
+   }
+--- a/system-error.hpp
++++ b/system-error.hpp
+@@ -23,6 +23,8 @@
+ #include <string>
+ #include <cstring>
+ 
++namespace Mapson {
```

Comment:
you could've also renamed `system_error` to `mapson_system_error` which is slightly less invasive (no ADL, friend lookup etc) but your choice